### DISCUSSION TITLE
calling save() on a model setup with table(false)

### DISCUSSION THREAD

**Discussion by mikegrogan** [wheels-dev/wheels#1213] [Q&A] [answered]
Working through a new scenario that I haven't dealt with before. I'm working on a small multi-page form that is just saving to the session scope. No table involved. I still want to validate certain fields. I was thinking I'd have to do this all in the controller, but then I remembered I could do table(false) in the model. If I call the save() i can get it to return errors like I want. However, if everything is valid, it is actually trying to call the save into a table, and throws an error. Is there a way to get around this? I don't see any parameter on the save function to do this. I thought if table(false) was set it would no to ignore the save basically. Or is there another way to just call the validation w/o the save call? I'm still on Wheels 1.4.5 fyi.

if (myobject.save()){
      // errors because it's actually trying to save to a table.
    }
    else{
      writeDump(var = myobject.allerrors(), abort = true);
    }

**Comment by dbelanger** [comment, marked as answer]
You are correct.  `.save()` won't work because it's a tableless model however if you use [valid()](https://api.cfwheels.org/v1.4.5#valid) you should get the results you're looking for!  It's the same process but without the save.

**Reply by mikegrogan** [reply]
This is exactly what I was looking for! Thanks. I knew I was missing something simple.

**Comment by adamcameron** [comment]
> .save() won't work because it's a tableless model 

I'm a bit unclear here. @mikegrogan should still be able to implement his own save method on a "tableless" model, and implement some other sort of persistence implementation... in this case "session persistence".

That would work, right?

One should not throw the whole "persisted object" paradigm out just because the persistence is not in a DB. 

If it's a tableless model, calling `save` should just either do nothing (not ideal), or throw something like a PersistenceTierNotImplementedException or something: make it clear that one actually needs to implement persistence if one actively doesn't want to use the baked-in persistence implementation.

The answer should not be "ah just bake yer persistence into yer controller instead". It's not the controller's job to do that.
